core/internal/api/api.go

```diff
@@ -210,10 +213,18 @@ func (backend *Backend) NewClient(opts ClientOptions) Client {
 		)
 	}
 
+	transport := retryableHTTP.HTTPClient.Transport.(*http.Transport)
```

Comment:
type-casting feels wrong here. I believe retryablehttp has a function to create a clean Transport; can we use that?